src/components/application_manager/test/commands/mobile/set_app_icon_test.cc

```diff
@@ -144,6 +160,515 @@ TEST_F(SetAppIconRequestTest, OnEvent_UI_UNSUPPORTED_RESOURCE) {
+  EXPECT_EQ(expect_result_code, actual_result_code);
+}
+
+TEST_F(SetAppIconRequestTest,
+       Run_SyncFileNameWithForbiddenSymbols_InvalidData) {
+  (*msg_vr)[am::strings::msg_params][am::strings::sync_file_name] =
+      "sync_file/";
+
+  utils::SharedPtr<SetAppIconRequest> req =
+      CreateCommand<SetAppIconRequest>(msg_vr);
+
+  MockAppPtr mock_app = CreateMockApp();
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  MessageSharedPtr result;
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _))
+      .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+
+  req->Run();
+
+  const mobile_apis::Result::eType expect_result_code =
+      mobile_apis::Result::INVALID_DATA;
+  const mobile_apis::Result::eType actual_result_code =
+      static_cast<const mobile_apis::Result::eType>(
+          (*result)[am::strings::msg_params][am::strings::result_code].asInt());
+  EXPECT_EQ(expect_result_code, actual_result_code);
+}
+
+TEST_F(SetAppIconRequestTest, Run_NonExistentFile_InvalidData) {
+  utils::SharedPtr<SetAppIconRequest> req =
+      CreateCommand<SetAppIconRequest>(msg_vr);
+
+  MockAppPtr mock_app = CreateMockApp();
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+
+  MockApplicationManagerSettings mock_app_mngr_settings;
+  EXPECT_CALL(app_mngr_, get_settings())
+      .WillRepeatedly(ReturnRef(mock_app_mngr_settings));
+
+  const std::string app_storage_folder = "./storage";
+  EXPECT_CALL(mock_app_mngr_settings, app_storage_folder())
+      .WillRepeatedly(ReturnRef(app_storage_folder));
+
+  const std::string app_folder = "Test_app";
+  EXPECT_CALL((*mock_app), folder_name()).WillRepeatedly(Return(app_folder));
+
+  MessageSharedPtr result;
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _))
+      .WillOnce(DoAll(SaveArg<0>(&result), Return(true)));
+
+  req->Run();
+
+  const mobile_apis::Result::eType expect_result_code =
+      mobile_apis::Result::INVALID_DATA;
+  const mobile_apis::Result::eType actual_result_code =
+      static_cast<const mobile_apis::Result::eType>(
+          (*result)[am::strings::msg_params][am::strings::result_code].asInt());
+  EXPECT_EQ(expect_result_code, actual_result_code);
+}
+
+TEST_F(SetAppIconRequestTest,
+       CopyToIconStorage_ProtocolVersionLessThanSupported_IconCopyingSkipped) {
+  utils::SharedPtr<SetAppIconRequest> req =
+      CreateCommand<SetAppIconRequest>(msg_vr);
+
+  MockAppPtr mock_app = CreateMockApp();
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillRepeatedly(Return(mock_app));
+
+  MockApplicationManagerSettings mock_app_mngr_settings;
+  EXPECT_CALL(app_mngr_, get_settings())
+      .WillOnce(ReturnRef(mock_app_mngr_settings));
+
+  const std::string app_storage_folder = "./storage";
+  EXPECT_CALL(mock_app_mngr_settings, app_storage_folder())
+      .WillRepeatedly(ReturnRef(app_storage_folder));
```

Comment:
@BSolonenko please use `.WillOnce`  if this expectation will be triggered only once. Check all expectations in this test